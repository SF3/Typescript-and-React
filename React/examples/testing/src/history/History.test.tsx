import React from "react";
import {History} from "./History";
import {render} from "@testing-library/react";
import {fireEvent} from "@testing-library/dom";

const clearMock = jest.fn();

beforeEach(() => {
    jest.resetAllMocks();
})

describe("Renders History Correctly", () => {
    it("Renders correct snapshot with empty log", () => {
        const target = render(
            <History log={[]} clearLog={clearMock}/>
        );

        expect(target.container.firstChild).toMatchSnapshot();
    });

    it("Renders correctly with log", () => {
        const target = render(
            <History log={["First", "Second"]} clearLog={clearMock}/>
        );

        expect(target.container.firstChild).toMatchSnapshot();
    });

    it("Calls clear function when button clicked", () => {
        const target = render(
            <History log={[]} clearLog={clearMock}/>
        );

        fireEvent.click(target.getByTestId('clear-button'));

        expect(clearMock).toHaveBeenCalled();
    });
});
