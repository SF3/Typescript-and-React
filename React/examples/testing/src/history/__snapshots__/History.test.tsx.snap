// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders History Correctly Renders correct snapshot with empty log 1`] = `
<div
  class="well border bg-light p-3"
>
  <div>
    <h1>
      History
    </h1>
    <button
      class="btn btn-primary"
      data-testid="clear-button"
    >
      Clear
    </button>
  </div>
  <ul
    id="historyList"
  />
</div>
`;

exports[`Renders History Correctly Renders correctly with log 1`] = `
<div
  class="well border bg-light p-3"
>
  <div>
    <h1>
      History
    </h1>
    <button
      class="btn btn-primary"
      data-testid="clear-button"
    >
      Clear
    </button>
  </div>
  <ul
    id="historyList"
  >
    <li>
      First
    </li>
    <li>
      Second
    </li>
  </ul>
</div>
`;
